File: habit-tracker/src/main/kotlin/com/habit/habit_tracker/dto/logs/weekly/WeeklyHabitLogResponse.kt
package com.habit.habit_tracker.dto.logs.weekly

import com.fasterxml.jackson.annotation.JsonInclude;
import java.time.LocalDate;

@JsonInclude(JsonInclude.Include.NON_NULL)
data class WeeklyHabitLogResponse(
    val id: Long,
    val habitId: Long,
    val weeklyGoal: Int?,
    val dailyGoal: Int?,
    val minutesDone: Int,
    val weeklyImbalance: Int?,
    val notes: String?,
    val weekStart: LocalDate,
    val weekEnd: LocalDate
)

File: client/src/types/dto/logs/response/weekly/WeeklyHabitLogResponse.ts
export interface WeeklyHabitLogResponse {
    id: number;
    habitId: number;
    weeklyGoal?: number | null;
    dailyGoal?: number | null;
    minutesDone: number;
    weeklyImbalance?: number | null;
    notes?: string | null;
    weekStart: string;
    weekEnd: string;
}